zjit/src/hir.rs

```diff
@@ -2003,9 +2013,12 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
     while let Some((incoming_state, block, mut insn_idx)) = queue.pop_front() {
         if visited.contains(&block) { continue; }
         visited.insert(block);
-        let mut state = if insn_idx == 0 { incoming_state.clone() } else {
+        let (self_param, mut state) = if insn_idx == 0 {
+            (fun.blocks[fun.entry_block.0].params[0], incoming_state.clone())
```

Comment:
```suggestion
            (fun.blocks[fun.entry_block.0].params[SELF_PARAM_IDX], incoming_state.clone())
```

Per Kevin's recommendation, this might be a good idea. Sorry I forgot it in my last note